/// Files: fs/fuse/cuse.c
/// Fix: 2c5816b4beccc8ba709144539f6fdd764f8fa49c
/// Fixes: cc080e9e9be16ccf26135d366d7d2b65209f1d56

virtual detect

@fuse_dev_alloc exists@
identifier cc;
@@

cuse_channel_open(...)
{
	...
	fuse_dev_alloc(&cc->fc)
	...
}

@err depends on fuse_dev_alloc exists@
identifier cc, inode, file;
position p;
@@

* cuse_channel_release@p(struct inode *inode, struct file *file)
{
	... when != fuse_conn_put(&cc->fc);
	fuse_dev_release(inode, file)
	...
}

@script:python depends on detect@
p << err.p;
@@

coccilib.report.print_report(p[0], 'ERROR: CVE-2015-1339')
